site/content/blog/2019-12-18-kubecon-na-2019.md

```diff
@@ -0,0 +1,70 @@
+---
+
+It wasn’t my first time flying. It also wasn’t the first time I had arrived at the airport early enough to require multiple cups of coffee. As I sat at my gate, however, I did find myself wondering for the first time: had I ever heard the words “lambda,” “stateful set,” and “yaml” while waiting to board?
+
+You see, a bunch of us were coming from all over to attend KubeCon + CloudNativeCon North America 2019. Now that the conference has come and gone, let’s dive into some of our takeaways.
+
+#### A container-sized intro
+
+Every year since the [first Kubernetes GitHub commit](https://github.com/kubernetes/kubernetes/commit/2c4b3a562ce34cddc3f8218a2c4d11c7310e6d56) in 2014, the project has needed less of an introduction. At its core, Kubernetes is an open source platform for running containerized workloads and services. With thousands of unique contributors and the support of some very large organizations, it has matured to the point where it runs varied complex workloads all over the world. This is, partly, due to the fact that Kubernetes is extensible in many ways and can be tailored to provide rich interactions and support for specific use-cases. We at Mattermost also find this incredibly valuable.
+
+With all that in mind, KubeCon is a conference where the community around Kubernetes comes together to talk about what is next, share lessons learned, compare experiences using the platform, and to simply connect.
+
+#### Kubernetes is not slowing down
+
+One of the most difficult aspects of attending Kubecon can simply be deciding which talks to attend. Nearly every time slot had a diverse set of topics which demonstrates that the platform is still being enhanced in nearly every technical direction. Everything from storage, networking, security, performance, monitoring, observability, and extensibility continues to change every year to meet new demands. There were also multiple amazing talks about how _not_ to do things, and why -- which are sometimes more useful than hearing what you _should_ be doing.
+
+{{< youtube FrQ8Lwm9_j8 >}}
+<br/>
+
+Additionally, version 1.16 of Kubernetes was recently released. Despite the platform regularly receiving quarterly minor version upgrades, a lot has been packed into the latest edition.
+
+In particular, we were excited that custom resources reached general availability in 1.16. Although this functionally doesn’t mean much, it shows that extensibility really is an important aspect of the platform. To break it down a bit, a custom resource definition allows the users of Kubernetes platforms to define their own specifications for resources. These specifications are then treated as normal Kubernetes API objects. This allows for something like a database, monitoring service, backend framework, etc. to be easily provisioned in a cluster based on these definitions. We use this to define Mattermost installations for our [operator to work with](https://github.com/mattermost/mattermost-operator).
```

Comment:
```suggestion
In particular, we were excited that custom resources reached general availability in 1.16. 

A custom resource definition allows the users of Kubernetes platforms to define their own specifications for resources. These specifications are then treated as normal Kubernetes API objects. This allows for something like a database, monitoring service, backend framework, etc. to be easily provisioned in a cluster based on these definitions. We use this to define Mattermost installations for our [operator to work with](https://github.com/mattermost/mattermost-operator). Although functionally this doesn’t mean much, it shows that extensibility really is an important aspect of the platform. 
```

Reply:
It sounds like custom resource definition is functionally useful. I'm wondering if there's a better way to describe it? Maybe something like "although this doesn't impact the day-to-day functionality..."?

Reply:
Sorry, I should maybe clarify, but the custom resource definitions have been around for a long time in alpha/beta. Kubernetes just likes to sit on its improvements for a long time before considering them "GA" level.